charon/src/bin/charon-driver/driver/config_mutation.rs

```diff
@@ -53,3 +54,18 @@ pub fn disabled_mir_passes(config: &mut Config) {
             .push((pass.to_owned(), false));
     }
 }
+
+/// Don't even try to codegen. This avoids errors due to checking if the output filename is
+/// available (despite the fact that we won't emit it because we stop compilation early).
+pub fn no_codegen(config: &mut Config) {
+    config.opts.unstable_opts.no_codegen = true;
+
+    // Original:
+    // compiler_args.push("--emit=metadata".to_string());
+    // FIXME: [`OutputTypes`] doesn't expose mutation methods, so here construct a new
+    // value to replace the whole output_types. This discards all outputs analyzed from
+    // cli args, so not sure if this breaks something.
```

Comment:
You can do something like `OutputTypes::new(config.opts.output_types.iter().cloned().chain([(OutputType::Metadata, None)]).collect_vec().as_slice())`.

Reply:
Done.

Small adjustment: `config.opts.output_types.iter()` yields `(&K, &V)`, so it needs to clone each element to get owned values.